src/lib/array.jl

```diff
@@ -124,6 +124,16 @@ end
   end
 end
 
+@adjoint repeat(x::AbstractVector, m::Integer) =
+   repeat(x, m), ȳ -> (dropdims(sum(reshape(ȳ, length(x), :); dims=2); dims=2), nothing)
```

Comment:
do we need this method? Doesn't the VecOrMat version cover both?

Reply:
Here's a test which fails without it:
```
gradient(x -> sum(1 .+ repeat(x, 2)::Vector), ones(2)) == ([2,2],)
```

Reply:
right, although I fail to see why. Without the type cast it works, and the return from `repeat(x, 2)` for x vector is already a vector, so the cast is useless :confused:

Reply:
No, the return from `repeat(x, 2)` is not a vector here, because `@adjoint function repeat(x::AbstractVecOrMat, ...` replaces it with `repeat(x, 2, 1)`.